### PR TITLE
Redux Configured

### PR DESCRIPTION
**Pull request by habitoz**
redux added and configured according to the given specification i have added the following files.
* created redux folder
* created booksSlice.js inside redux/books folder
* created categoriesSlice.js inside redux/categories folder
* created store.js
* updated main.js to add provider

### CODE REVIEW

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
> Hi @habitoz  ,
> 
> Great job for implementing  some of the Project Requirement, but there are some changes that need to be implemented in order to proceed to the next project 👍
> 
> ## Highlights 👏🏻
> ✔️ Created actions , payloads and reducers ✔️ You used correct Git Hub-Flow. ✔️ Good README ✔️ Good Commit Messages
> 
> almost 

> ## Required Changes ♻️
> _Check the comments below the review._
> 

> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    remove(store, action) {
+      return store.filter((book) => book.id !== action.payload);
+    },
+    add(store, action) {
+      store.push(action.payload);
+    },
+  },
+});
+
+export default booksSlice.reducer;
```
Comment (bobb-Rob) [inline src/redux/books/booksSlice.js:18]: - [ ] You have not exported the `add` and `remove` action like so 👇🏾 

```javascript
export const { add, remove } = booksSlice.actions
```

For better readability your reducers should be called `addBook` and `removeBook`

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  name: 'categories',
+  initialState,
+  reducers: {
+    status() {
+      return 'Under construction';
+    },
+  },
+});
+
+export default categoriesSlice.reducer;
```
Comment (bobb-Rob) [inline src/redux/categories/categoriesSlice.js:15]: - [ ] Similarly, you should export the `status` action

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @habitoz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ All linters checks are ok
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**